docs/rst/tutorials/micro/handle_reconnections/handle_reconnections.rst

```diff
@@ -0,0 +1,136 @@
+.. _tutorials_micro_handle_reconnections:
+
+Reconnections and liveliness
+============================
+
+.. contents:: Table of Contents
+    :depth: 1
+    :local:
+    :backlinks: none
+
+This tutorial aims at providing insight on mechanism for handling reconnections between the micro-ROS Agent and Client and assert the connection liveliness.
+
+This features can be useful in multiple situations when the Agent or the Client are no longer available or the communication channel between them is broken. Those situations can be, among others:
+
+- Transport faults: Detect transport fault or disconnections and act accordingly.
+- Agent or Client restarts: Make the system resilient to a possible restart on the MCU or Agent side.
+- Client plug and play: Useful on systems with multiple MCU boards that are attached on demand.
+
+Client side: Ping API
+---------------------
+
+micro-ROS Client default rmw layer `rmw_microxrcedds_c <https://github.com/micro-ROS/rmw_microxrcedds>`_ offers a ping utility to test the connection with the Agent. This ping utility can be used at any stage of the micro-ROS application, allowing users to check the Agent availability before attempting to initialize a micro-ROS session or create any entities.
+
+Usage details on this utility can be found in the :ref:`Middleware API: Ping Agent <tutorials_micro_user_middleware_ping>` tutorial.
+
+State machine approach
+^^^^^^^^^^^^^^^^^^^^^^
+
+Reconnections can be handle manually on the micro-ROS Client side using the Agent ping functionality with the following sequence:
+
+1. Wait until Agent is reachable.
+2. Create required micro-ROS entities.
+3. Use the micro-ROS API as usual: spin entities, publish, etc.
+4. When a failure is detected in steps 2 or 3, destroy the created entities and go back to the first step.
```

Comment:
is there an easy way to add sequence or activity diagrams to rst?? like mermaid on githuib markdown, I feel that an activity diagram for this sequence would be nice but only if its easy to include

Reply:
Good idea, maybe another PR